backend/tests/e2e/test_pause_resume_e2e.py

```
@@ -0,0 +1,629 @@
+        # Verify stage-level token counts
+        if total_tokens > 0:
+            assert actual_stage['stage_input_tokens'] == total_input_tokens, \
+                f"Stage '{stage_name}' ({stage_key}) stage_input_tokens mismatch: expected {total_input_tokens}, got {actual_stage['stage_input_tokens']}"
+            assert actual_stage['stage_output_tokens'] == total_output_tokens, \
+                f"Stage '{stage_name}' ({stage_key}) stage_output_tokens mismatch: expected {total_output_tokens}, got {actual_stage['stage_output_tokens']}"
+            assert actual_stage['stage_total_tokens'] == total_tokens, \
+                f"Stage '{stage_name}' ({stage_key}) stage_total_tokens mismatch: expected {total_tokens}, got {actual_stage['stage_total_tokens']}"
+        
+        # Verify conversation structure if expected
+        if 'expected_conversation' in expected_stage:
+            print(f"   🔍 Validating conversation history (proving restoration from pause)...")
```

Ruff (0.14.4)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)